### ISSUE TITLE
vSphere cloudprovider does not sanitise region or failuredomain, so nodes fail to register

### ISSUE THREAD

**Issue by jakexks** [kubernetes/kubernetes#33576]
**Kubernetes version**

```
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:16:57Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:10:32Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- vSphere cloud provider
- **OS** CoreOS 1122.2.0 (MoreOS)
- **Kernel** Linux master0 4.7.0-coreos #1 SMP Tue Sep 6 14:39:20 UTC 2016 x86_64 Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz GenuineIntel GNU/Linux

**What happened**:

Kubelet fails to register nodes when the vSphere data center name or failure zone contain invalid characters.

```
E0927 13:28:03.940239   19470 kubelet_node_status.go:88] Unable to register node "worker0" with API server: Node "worker0" is invalid: metadata.labels: Invalid value: "my Datacenter": must match the regex (([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])? (e.g. 'MyValue' or 'my_value' or '12345')
```

**What you expected to happen**:

Cloud provider should sanitise invalid names

**How to reproduce it** (as minimally and precisely as possible):

Create a vSphere data center called "my Datacenter"

**Anything else do we need to know**:

PR Incoming

**Comment by BaluDontu** [comment]
This issue is fixed in release 1.4.7. Currently we decided to have no support for regions or fault domains as it is not a fully implemented functionality. 

So in vsphere CP, you will see no more references to regions and fault domains.

**Comment by kerneltime** [comment]
@jakexks can you give 1.4.7 a try. More info here http://kubernetes.io/docs/getting-started-guides/vsphere/
